src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+import java.util.ArrayList;
+
+public class Ui {
+
+    /**
+     * Returns the welcome message of the duke chat bot.
+     * 
+     * @return Welcome Message
+     */
+    public static String WelcomeMessage() {
+        String separator = "    ____________________________________________________________";
+        return separator + "\n" + "     Hello! I'm Duke" + "\n" + "     What can I do for you?"  + "\n" + separator;
+    }
+
+    /**
+     * Returns updated user task list and prints feedback to the user based on their input
+     * 
+     * @param taskList list of user's tasks
+     * @param input user input
+     * @return updated task list
+     */
+    public ArrayList<Task> run(ArrayList<Task> taskList, String input) {
+        ArrayList<Task> list = taskList;
+        String separator = "    ____________________________________________________________";
+        try {
+            String parsedInput = Parser.parse(input);
+            if (parsedInput.equals("list")) {
+                if (list.size() == 0) {
+                    System.out.println(separator);
+                    System.out.println("     You have no tasks in your list!");
```

Comment:
A UI component that can be abstracted to a generic method to format responses. Since these lines of code are used extensively in the run method, abstracting the method will reduce code re-write .